test/unit/api_spec.js

```diff
@@ -979,6 +979,15 @@ describe('api', function() {
       expect(viewport.width).toEqual(1262.835);
       expect(viewport.height).toEqual(892.92);
     });
+    it('gets viewport respecting "dontFlip" argument', function () {
+      const scale = 1.5;
+      const rotation = 135;
+      const flipped = page.getViewport(scale, rotation);
+      const unflipped = flipped.clone({ dontFlip: true, });
+      const notFlippedFromTheStart = page.getViewport(scale, rotation, true);
+
+      expect(notFlippedFromTheStart.transform).toEqual(unflipped.transform);
+    });
```

Comment:
Personally I'd actually like to see somewhat more complete tests here, to ensure that what the API returns is *exactly* the expected result (especially since this unit-test is the only place the code-path used), given that you're *only* checking the `transform` property and nothing else here.

Hence I'd prefer something along the following lines instead.
```javascript
let viewport = page.getViewport(scale, rotation);
let dontFlipViewport = page.getViewport(scale, rotation, true);

expect(viewport.transform).toEqual([/* explicit transform array here */]);
expect(dontFlipViewport.transform).toEqual([/* explicit transform array here */]);

expect(dontFlipViewport).toEqual(viewport.clone({ dontFlip: true, }));
```

Reply:
Pls check now, I pushed changes.